### PR TITLE
Add role for ManageIQ/CloudForms deployment

### PR DESCRIPTION
**Pull request by machacekondra**
Add role for ManageIQ/CloudForms deployment

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina

**Comment by machacekondra** [comment]
@mgoldboi @oourfali

File: roles/ovirt-manageiq/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+|--------------------|-------------------|-------------------------------------------------|
+| ovirt_user         | UNDEF             | Define the user to be used to access oVirt/RHV. |
+| ovirt_password     | UNDEF             | Define password of the 'ovirt_user'.            |
+| ovirt_fqdn         | UNDEF             | Define the FQDN of the oVirt/RHV.               |
+| ovirt_ca           | UNDEF             | Define the path to the CA of the oVirt/RHV.     |
+
+Virtual Machine variables:
+| Name               | Default value     |  Description                                 |
+|--------------------|-------------------|----------------------------------------------|
+|  miq_vm_name       | manageiq_euwe     | Define the name of the ManageIQ/CloudForms VM in oVirt/RHV.|
```
Comment (oourfali) [inline roles/ovirt-manageiq/README.md]: remove the extra space.

File: roles/ovirt-manageiq/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  register: downloaded
+
+- name: Check file type
+  command: "/usr/bin/file {{ miq_image_path | quote }}"
+  changed_when: false
+  register: filetype
+
+- name: Fail if image is not qcow
+  fail:
+    msg: "The downloaded file is not valid QCOW file."
```
Comment (oourfali) [inline roles/ovirt-manageiq/tasks/main.yml]: "a valid"

File: roles/ovirt-manageiq/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      type: server
+      disks:
+        - id: "{{ ovirt_disk.id }}"
+          bootable: true
+      cloud_init: "{{ miq_vm_cloud_init | default(omit) }}"
+      nics: "{{ miq_vm_nics }}"
+    when: ovirt_vms | length == 0
+    register: create_vm
+
+  - name: Wait for VM IP
```
Comment (oourfali) [inline roles/ovirt-manageiq/tasks/main.yml:77]: does the image include an ovirt guest agent?

**Review by oourfali — COMMENTED** [review]
Also I wonder whether we should also add DWH properties (so that capacity and utilization will be configured as well).
Do we have these details? Shall we ask the user fo fill them in?

File: roles/ovirt-manageiq/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+oVirt/RHV ManageIQ/CloudForms deployment
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: I'd rephrase as "ManageIQ/CloudForms deployment on top of oVirt/RHV"

File: roles/ovirt-manageiq/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+You also have to use your CA certificate of the engine.
+You must configure the `ovirt_ca` varible with the path to the CA.
+
+Role Variables
+--------------
+
+QCOW varibles:
+```
+ - miq_qcow_url - Define the URL where the ManageIQ/CloudForms qcow is stored (default: http://releases.manageiq.org/manageiq-ovirt-euwe-1.ova).
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: Here we have a link to OVA, below you are using link to qcow2, so shouldn't use single term everywhere and just mention in the doc that both OVA or qcow2 formats can be used?

File: roles/ovirt-manageiq/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+| ovirt_ca           | UNDEF             | Define the path to the CA of the oVirt/RHV.     |
+
+Virtual Machine variables:
+| Name               | Default value     |  Description                                 |
+|--------------------|-------------------|----------------------------------------------|
+|  miq_vm_name       | manageiq_euwe     | Define the name of the ManageIQ/CloudForms VM in oVirt/RHV.|
+| miq_vm_cluster     | Default           | Define the cluster of the virtual machine    |
+| miq_vm_memory      | 6GiB              | Define the system memory of the virtual machine    |
+| miq_vm_cpu         | 2                 | Define the number of virtual machine CPU cores   |
+| miq_vm_os          | rhel_7x64         | Define the operating system of the virtual machine |
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: The default is upstream, should we should use CentOS instead of RHEL

File: roles/ovirt-manageiq/README.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+Note that for passwords you should be using Ansible vault.
+
+```yaml
+    - name: Deploy ManageIQ to oVirt engine
+      hosts: localhost
+      gather_facts: no
+
+      vars:
+        ovirt_fqdn: ondra.local
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: ovirt-engine.example.com

File: roles/ovirt-manageiq/README.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        ovirt_password: 123456
+        ovirt_username: admin@internal
+
+        miq_vm_name: manageiq_euwe
+        miq_qcow_url: http://releases.manageiq.org/manageiq-ovirt-euwe-3.qc2
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: Does ManageIQ provide some link to latest release, so we won't need to update this often?

File: roles/ovirt-manageiq/README.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        miq_vm_cluster: mycluster
+        miq_vm_cloud_init:
+          user_name: root
+          root_password: securepassword
+
+      roles:
+        - machacekondra.ovirt-manageiq
```
Comment (mwperina) [inline roles/ovirt-manageiq/README.md]: Shouldn't there be only "ovirt.manageiq" or "ovirt-ansible.ovirt-manageiq"?

File: roles/ovirt-manageiq/defaults/main.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+miq_password: smartvm
+miq_provider_name: RHV provider
+
+### oVirt/RHV ###
+# VM variables:
+miq_vm_name: manageiq_euwe
+miq_vm_cluster: Default
+miq_vm_memory: 6GiB
+miq_vm_cpu: 2
+miq_vm_os: rhel_7x64
```
Comment (mwperina) [inline roles/ovirt-manageiq/defaults/main.yml]: CentOS?

**Review by mwperina — CHANGES_REQUESTED** [review]

**Comment by mgoldboi** [comment]
would be great if we can add the option to configure additional providers as well as part of the deployment/ configuration stage.
for the use case this CF engine is managing a different provider than oVirt. Openshift for example can be nice part of this story.

**Comment by mwperina** [comment]
> would be great if we can add the option to configure additional providers as well as part of the 
> deployment/ configuration stage.
> for the use case this CF engine is managing a different provider than oVirt. Openshift for example can 
> be nice part of this story.

Not sure if this part of this role, isn't a role configuring OpenShift provider in MiQ part of OpenShift roles, so users could reuse it?

File: roles/ovirt-manageiq/tasks/main.yml (hunk 3 above)
Comment (machacekondra) [reply roles/ovirt-manageiq/tasks/main.yml:77]: The ManageIQ includes them, I didn't try the CloudForms.

File: roles/ovirt-manageiq/README.md (hunk 6 above)
Comment (machacekondra) [reply roles/ovirt-manageiq/README.md]: There is no such option.

File: roles/ovirt-manageiq/README.md (hunk 8 above)
Comment (machacekondra) [reply roles/ovirt-manageiq/README.md]: I didn't find any.

File: roles/ovirt-manageiq/README.md (hunk 9 above)
Comment (machacekondra) [reply roles/ovirt-manageiq/README.md]: Yes.

File: roles/ovirt-manageiq/README.md (hunk 5 above)
Comment (machacekondra) [reply roles/ovirt-manageiq/README.md]: Added the qcow only.

File: roles/ovirt-manageiq/README.md (hunk 6 above)
Comment (mwperina) [reply roles/ovirt-manageiq/README.md]: Ahh, so this is oVirt specific, we need to set VM OS to RHEL, even though CentOS is used for ManageIQ VM.

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks!

**Comment by machacekondra** [comment]
Added Openshift and DWH support.

**Comment by mykaul** [comment]
Openshift is unrelated to Manage IQ?

**Comment by machacekondra** [comment]
It's related, you can for example manage Openshift in RHV all via ManageIQ. This was requested by Moran.

**Comment by mykaul** [comment]
Makes little sense to me.
I see 2-3 roles:
- Install ManageIQ
- Install OpenShift
- Connect the two.

It's not natural to see it in a single role.

**Comment by oourfali** [comment]
I also remember we agreed to keep openshift out of the scope of this role.
Am I wrong?

**Comment by machacekondra** [comment]
It's optional. It's just one API call, so no problem to remove it.

Yes, that could be separated to those roles, I think better it would be if this role just would deploy ManageIQ, register RHV and then the user can use some ManageIQ modules( if any ) and register other providers in post_tasks.

I will split it.

**Review by mwperina — CHANGES_REQUESTED** [review]
Let's remove Openshift part from this role, it's should be of a different role. Other than that I think it's ready to merge

**Review by mwperina — APPROVED** [review]
👍
